src/data_storage/validators.py

```diff
@@ -0,0 +1,59 @@
+from itertools import chain
+
+from utilities.exceptions import (
+    ExistsCircleValidationError,
+    RootDoesNotExistValidationError,
+    UnconnectedNodeValidationError,
+    UnrelatedReferenceValidationError,
+)
+
+
+def deep_first_search(graph, current_vertex, color=None):
```

Comment:
`True` means that we don't have cycles, right? 
```suggestion
def deep_first_search(graph, current_vertex, color=None) -> bool:
```